openml/datasets/dataset.py

```diff
@@ -295,8 +295,11 @@ def __eq__(self, other):
             "version",
             "upload_date",
             "url",
-            "dataset",
             "data_file",
+            "_dataset",
+            "data_pickle_file",
+            "data_feather_file",
+            "feather_attribute_file",
```

Comment:
I'd like to see these defined as a separate set (e.g. `cache_fields`), because these are not server fields.